command/agent/fs_endpoint.go

```diff
@@ -466,8 +473,8 @@ func (s *StreamFramer) Send(file, fileEvent string, data []byte, offset int64) e
 	// If we are not running, return the error that caused us to not run or
 	// indicated that it was never started.
 	if !s.running {
-		if s.Err != nil {
-			return s.Err
+		if s.err != nil {
```

Comment:
Should this be `s.Err()`? Otherwise add a comment about why this can be accessed directly.

Reply:
Same for line 477

Reply:
This method holds the appropriate lock (L470)